core/primitives/stream.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_CORE_PRIMITIVES_STREAM_HPP
+#define FILECOIN_CORE_PRIMITIVES_STREAM_HPP
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <deque>
+
+namespace fc::core::primitives {
+
+  /**
+   * Base class for representing data streams
+   */
+  class Stream {
```

Comment:
I think we should split encoding and decoding streams into different files

Reply:
Spitted to primitives::DecodeStream and primitives::EncodeStream